ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -699,7 +699,15 @@ public void deleteStoreFiles() throws StoreException, IOException {
       File swapSegmentTempFile = new File(dataDir, fileName);
       diskSpaceAllocator.free(swapSegmentTempFile, config.storeSegmentSizeInBytes, storeId, true);
     }
-    // Step 2: delete all files
+    // Step 2: if segmented, delete remaining store segments in reserve pool
+    if (log.isLogSegmented()) {
+      Map<String, Map<Long, Long>> storeSegmentRequirement = new HashMap<>();
+      Map<Long, Long> segmentSizeAndNum = new HashMap<>();
+      segmentSizeAndNum.put(log.getSegmentCapacity(), log.getRemainingUnallocatedSegments());
+      storeSegmentRequirement.put(storeId, segmentSizeAndNum);
+      diskSpaceAllocator.deleteExtraSegments(storeSegmentRequirement, false);
```

Comment:
I would recommend removing the isInitialize flag (and make deleteExtraSegments private) and adding a new method `deleteAllSegmentsForStoreIds(List<String> storeIds)` for the non-init use case